tests/testthat/setup.R

```diff
@@ -110,3 +47,57 @@ if (nchar(Sys.getenv("GITLAB_PAT_PUBLIC")) > 0) {
     }
   })
 }
+
+test_mocker <- Mocker$new()
```

Comment:
Locally some tests failed; 

```
══ Failed tests 
════════════════════════════════════════════════════════════════
── Error ('test-get_files_structure-GitLab.R:176:3'): get_files_structure_from_orgs pulls files structure for all repositories in orgs ──
<purrr_error_indexed/rlang_error/error/condition>
Error in `purrr::map(private$orgs, function(org) {
    if (verbose) {
        user_info <- if (!is.null(pattern)) {
            glue::glue("Pulling files structure...[files matching pattern: '{pattern}']")
        }
        else {
            glue::glue("Pulling files structure...")
        }
        show_message(host = private$host_name, engine = "graphql", 
            scope = org, information = user_info)
    }
    graphql_engine$get_files_structure_from_org(org = org, repos = private$repos, 
        pattern = pattern, depth = depth, verbose = verbose, 
        progress = progress)
})`: i In index: 1.
Caused by error in `purrr::map()`:
i In index: 6.
Caused by error in `httr2::req_perform()`:
! HTTP 502 Bad Gateway.
Backtrace:
     ▆
  1. ├─gitlab_testhost_priv$get_files_structure_from_orgs(...) at test-get_files_structure-GitLab.R:176:3
  2. │ └─purrr::map(...)
  3. │   └─purrr:::map_("list", .x, .f, ..., .progress = .progress)
  4. │     ├─purrr:::with_indexed_errors(...)
  5. │     │ └─base::withCallingHandlers(...)
  6. │     ├─purrr:::call_with_cleanup(...)
  7. │     └─GitStats (local) .f(.x[[i]], ...)
  8. │       └─graphql_engine$get_files_structure_from_org(...)
  9. │         └─purrr::map(...)
 10. │           └─purrr:::map_("list", .x, .f, ..., .progress = .progress)
 11. │             ├─purrr:::with_indexed_errors(...)
 12. │             │ └─base::withCallingHandlers(...)
 13. │             ├─purrr:::call_with_cleanup(...)
 14. │             └─GitStats (local) .f(.x[[i]], ...)
 15. │               └─private$get_files_structure_from_repo(...)
 16. │                 └─private$get_files_tree_response(org = org, repo = repo, file_path = "")
 17. │                   └─self$gql_response(...)
 18. │                     └─private$perform_request(gql_query = gql_query, vars = vars)
 19. │                       └─... %>% httr2::req_perform()
 20. ├─httr2::req_perform(.)
 21. │ └─httr2:::handle_resp(req, resp, error_call = error_call)
 22. │   └─httr2:::resp_abort(resp, req, body, call = error_call)
 23. │     └─rlang::abort(...)
 24. │       └─rlang:::signal_abort(cnd, .file)
 25. │         └─base::signalCondition(cnd)
 26. ├─purrr (local) `<fn>`(`<ht2__502>`)
 27. │ └─cli::cli_abort(...)
 28. │   └─rlang::abort(...)
 29. │     └─rlang:::signal_abort(cnd, .file)
 30. │       └─base::signalCondition(cnd)
 31. └─purrr (local) `<fn>`(`<prrr_rr_>`)
 32.   └─cli::cli_abort(...)
 33.     └─rlang::abort(...)
── Failure ('test-get_files_structure-GitLab.R:203:3'): get_path_from_files_structure gets file path from files structure ──
typeof(file_path) (`actual`) not equal to "character" (`expected`).

`actual`:   "NULL"     
`expected`: "character"
── Failure ('test-get_files_structure-GitLab.R:204:3'): get_path_from_files_structure gets file path from files structure ──
length(file_path) > 0 is not TRUE

`actual`:   FALSE
`expected`: TRUE 
── Failure ('test-get_files_structure-GitLab.R:219:3'): get_files_structure pulls files structure for repositories in orgs ──
names(gl_files_structure_from_orgs) (`actual`) not equal to c("mbtests", "mbtestapps") (`expected`).

`actual` is NULL
`expected` is a character vector ('mbtests', 'mbtestapps')
── Failure ('test-get_files_structure-GitLab.R:235:3'): get_files_content makes use of files_structure ──
Snapshot of code has changed:
old vs new
  Code
    files_content <- gitlab_testhost_priv$get_files_content_from_orgs(file_path = NULL,
      host_files_structure = test_mocker$use("gl_files_structure_from_orgs"))
- Message
-   i I will make use of files structure stored in GitStats.

* Run `testthat::snapshot_accept('get_files_structure-GitLab')` to accept the change.
* Run `testthat::snapshot_review('get_files_structure-GitLab')` to interactively review the change.

[ FAIL 5 | WARN 0 | SKIP 0 | PASS 537 ]
```

Reply:
The reason here is server error. Nonetheless, shows, I did not mock this API responses.